docs/dev/zksync-v2/aa.md

```diff
@@ -32,26 +35,32 @@ By default, calling `estimateGas` adds a constant of `20000` to cover charging t
 
-- `validateTransaction` is mandatory and will be used by the system to determine if the AA logic agrees to proceed with the transaction. In case the transaction is not accepted (e.g. the signature is wrong) your AA should revert. In case the call to this method returns `true`, the AA is considered to accept the this transaction and will be charged with fee afterwards.
+- `validateTransaction` is mandatory and will be used by the system to determine if the AA logic agrees to proceed with the transaction. In case the transaction is not accepted 
+  (e.g. the signature is wrong) the transaction fails. In case the call to this method returns `true`, the implemented account logic is considered to accept the transaction and will be charged with fee afterwards.
 - `executeTransaction` is mandatory and will be called by the system after the fee is charged from the user. This function should perform the execution of the transaction.
-- `executeTransactionFromOutside`, technically, is not mandatory, but it is _highly encouraged_, since there needs to be some way, in case of priority mode (e.g. when the operator becomes malicious), to be able to start transactions from your AA from the outside (basically this is the fallback to the standard Ethereum approach, where an EOA starts transaction from your smart contract).
+- `executeTransactionFromOutside`, technically, is not mandatory, but it is _highly encouraged_, since there needs to be some way, in case of priority mode (e.g. when the operator becomes malicious), to be able to start transactions from your account from ``outside'' (basically this is the fallback to the standard Ethereum approach, where an EOA starts transaction from your smart contract).
 
-Note, that each of these methods accept the [Transaction](https://github.com/matter-labs/v2-testnet-contracts/blob/0e1c95969a2f92974370326e4430f03e417b25e7/l2/system-contracts/TransactionHelper.sol#L15) struct. While some of its fields are self-explanatory, there are also 6 `reserved` fields, the meaning of each will be defined by the transaction's type. We decided to not give these fields names, since they might be unneeded in some future transaction types. For now, the convention is:
+Note that each of these methods accept the [Transaction](https://github.com/matter-labs/v2-testnet-contracts/blob/0e1c95969a2f92974370326e4430f03e417b25e7/l2/system-contracts/TransactionHelper.sol#L15) struct. 
+While some of its fields are self-explanatory, there are also 6 `reserved` fields, the meaning of each is defined by the transaction's type. We decided to not give these fields names, since they might be unneeded in some future transaction types. For now, the convention is:
 
 - `reverted[0]` is the nonce.
-- `reserved[1]` is the `msg.value` that should be passed with the transaction.
+- `reserved[1]` is `msg.value` that should be passed with the transaction.
```

Comment:
WTF? Eth transfers and calls with non-zero value should always be done as formal calls to `ETHSimulator` contracts with specially crafted calldata. Bootloader must not try to glue-craft any special format of call to `ETHSimulator` by using `reserved[1]` and the "true" `to` and `calldata` fields

Reply:
@StanislavBreadless should check it again, but we did a lot of effort to avoid special-casing in the whole system in general for "execution" workflow - all the logic of the transaction is determined by `to` and `calldata` fields. `reserved` array was placed for purposes of future AA extensions, and not the "call logic" extensions

Reply:
Yes, discussed it with @shamatar, I created a follow-up task on it: 
https://linear.app/matterlabs/issue/SMA-1010/refactor-aatransaction-type-for-zero-copying

To be fully completed, this task requires product research. Until it is done, what you wrote @barakshani is true

Reply:
mmm..... I haven't really written anything, I am just editing the existing text